src/phantom.js

```diff
@@ -27,25 +28,40 @@ export default class Phantom {
      * Creates a new instance of Phantom
      *
      * @param args command args to pass to phantom process
-     * @param [config] configuration object
-     * @param [config.phantomPath] path to phantomjs executable
+     * @param [phantomPath] path to phantomjs executable
+     * @param [logger] object containing functions used for logging
+     * @param [logLevel] log level to apply on the logger (if unset or default)
      */
-    constructor(args = [], config = {}) {
+    constructor(args = [], {phantomPath = phantomjs.path, logger = defaultLogger, logLevel = defaultLogLevel} = {
+        phantomPath: phantomjs.path,
+        logger: defaultLogger,
+        logLevel: defaultLogLevel
+    }) {
         if (!Array.isArray(args)) {
             throw new Error('Unexpected type of parameters. Expecting args to be array.');
         }
-        if (!config || typeof config !== 'object') {
-            throw new Error('Unexpected type of parameters. Expecting config to be object.');
-        }
-
-        let phantomPath = typeof config.phantomPath === 'string' ? config.phantomPath : phantomjs.path;
 
         if (phantomPath === null) {
```

Comment:
It could be replaced with `typeof phantomPath !== 'string'` maybe, so it checks multiple cases at once.